test/e2e/kappcontroller/installation_namespace_test.go

```diff
@@ -0,0 +1,288 @@
+
+	"github.com/vmware-tanzu/carvel-kapp-controller/test/e2e"
+)
+
+func Test_AppInstallationNamespace(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	kubectl := e2e.Kubectl{t, env.Namespace, logger}
+	name := "app-installation-namespace"
+	installationNamespace := "installation-namespace"
+	installationNamespaceApp := "installation-namespace-app"
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", installationNamespaceApp})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	namespaceYAML := fmt.Sprintf(`---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s`, installationNamespace)
+
+	sas := e2e.ServiceAccounts{env.Namespace}
+	appYAML := fmt.Sprintf(`---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  installationNamespace: %s
+  fetch:
+  - inline:
+      paths:
+        file.yml: |
+          apiVersion: v1
+          kind: ConfigMap
+          metadata:
+            name: my-cm
+          data:
+            key: value
+  template:
+  - ytt: {}
+  deploy:
+  - kapp: {}`, name, env.Namespace, installationNamespace) + sas.ForClusterYAML()
+
+	kapp.RunWithOpts([]string{"deploy", "-a", installationNamespaceApp, "-f", "-"}, e2e.RunOpts{StdinReader: strings.NewReader(namespaceYAML)})
+
+	_, err := kapp.RunWithOpts([]string{"deploy", "-a", name, "-f", "-"}, e2e.RunOpts{StdinReader: strings.NewReader(appYAML)})
+	if err != nil {
+		t.Fatalf("Expected app deploy to succeed, it did not: %v", err)
+	}
+
+	_, err = kubectl.RunWithOpts([]string{"get", "configmap", "my-cm", "-n", installationNamespace}, e2e.RunOpts{AllowError: true, NoNamespace: true})
+	if err != nil {
+		t.Fatalf("Expected to find app resources in %s namespace, but did not: %v", installationNamespace, err)
+	}
+
+	_, err = kubectl.RunWithOpts([]string{"get", "configmap", name + ".app", "-n", env.Namespace}, e2e.RunOpts{AllowError: true, NoNamespace: true})
+	if err != nil {
+		t.Fatalf("Expected to find kapp app configmap in %s namespace, but did not: %v", env.Namespace, err)
+	}
+}
+
+func Test_PackageInstall_InstallationNamespace(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	kubectl := e2e.Kubectl{t, env.Namespace, logger}
+	name := "app-installation-namespace"
+	installationNamespace := "installation-namespace"
+	pkgiNamespace := "pkgi-namespace"
+	pkgiName := "test-pkgi"
+	appName := "test-app"
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	namespacesYAML := fmt.Sprintf(`---
+          apiVersion: v1
+          kind: Namespace
+          metadata:
+            name: %s
+          ---
+          apiVersion: v1
+          kind: Namespace
+          metadata:
+            name: %s
+`, installationNamespace, pkgiNamespace)
+
+	rbacYAML := fmt.Sprintf(`
```

Comment:
Any change we could move this to `service_accounts.go` into a new function? I feel like if we ever have to change this SA setup for the tests that this test is likely to get forgotten about.

Reply:
Done.